src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can not add country to DB, country = " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return Optional.ofNullable(
+                    session.get(Country.class, id)
+            );
+        } catch (Exception e) {
+            throw new DataProcessingException("Can not return movie by id, " + id, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can not return country by id, " + id, e);
```
Be carefull when copy-paste

Reply:
ok ;)